File: android/src/main/java/org/ligi/passandroid/ui/PassMenuOptions.java
package org.ligi.passandroid.ui;

import android.app.Activity;
import android.content.DialogInterface;
import android.content.Intent;
import android.support.v4.app.NavUtils;
import android.support.v7.app.AlertDialog;
import android.view.MenuItem;
import android.widget.CheckBox;

import org.ligi.axt.AXT;
import org.ligi.passandroid.App;
import org.ligi.passandroid.R;
import org.ligi.passandroid.Tracker;
import org.ligi.passandroid.maps.PassbookMapsFacade;
import org.ligi.passandroid.model.Pass;
import org.ligi.passandroid.model.PassStore;
import org.ligi.passandroid.model.Settings;

import java.io.File;

import javax.inject.Inject;

public class PassMenuOptions {

    @Inject
    PassStore passStore;

    @Inject
    Tracker tracker;

    @Inject
    Settings settings;

    public final Activity activity;
    public final Pass pass;

    public PassMenuOptions(final Activity activity, final Pass pass) {
        App.component().inject(this);
        this.activity = activity;
        this.pass = pass;
    }

    public boolean process(MenuItem item) {
        switch (item.getItemId()) {

            case R.id.menu_delete:
                tracker.trackEvent("ui_action", "delete", "delete", null);

                final AlertDialog.Builder builder = new AlertDialog.Builder(activity);
                builder.setMessage(activity.getString(R.string.dialog_delete_confirm_text));
                builder.setTitle(activity.getString(org.ligi.passandroid.R.string.dialog_delete_title));
                builder.setIcon(R.drawable.ic_warning_amber_36dp);

                final CheckBox sourceDeleteCheckBox = new CheckBox(activity);

                if (pass.getSource() != null && pass.getSource().startsWith("file://")) {
                    sourceDeleteCheckBox.setText(activity.getString(R.string.dialog_delete_confirm_delete_source_checkbox));
                    builder.setView(sourceDeleteCheckBox);
                }

                builder.setPositiveButton(activity.getString(R.string.delete), new DialogInterface.OnClickListener() {

                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (sourceDeleteCheckBox.isChecked()) {
                            new File(pass.getSource().replace("file://", "")).delete();
                        }
                        passStore.deletePassWithId(pass.getId());
                        if (activity instanceof PassViewActivityBase) {
                            final Intent passListIntent = new Intent(activity, PassListActivity.class);
                            NavUtils.navigateUpTo(activity, passListIntent);
                        } else if (activity instanceof PassListActivity) {
                            ((PassListActivity) activity).refreshPasses();
                        }
                    }

                });
                builder.setNegativeButton(android.R.string.no, null);

                builder.show();

                return true;

            case R.id.menu_map:
                PassbookMapsFacade.startFullscreenMap(activity, pass);
                return true;

            case R.id.menu_share:
                tracker.trackEvent("ui_action", "share", "shared", null);
                new AlertDialog.Builder(activity).setItems(new CharSequence[]{"OpenPass ( no Apple support yet)", "Passbook"},
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(final DialogInterface dialog, final int which) {
                                exportInFormat(which);
                            }
                        }).show();

                return true;

            case R.id.menu_edit:
                tracker.trackEvent("ui_action", "share", "shared", null);
                passStore.setCurrentPass(pass);
                AXT.at(activity).startCommonIntent().activityFromClass(PassEditActivity.class);
                return true;
        }
        return false;
    }

    private void exportInFormat(final int which) {
        final int passFormat = getPassFormat(which);
        new PassExportTask(activity, pass.getPath(), settings.getShareDir(), "share", true, passFormat).execute();
    }

    @PassExporter.PassFormat
    private int getPassFormat(int which) {
        switch (which) {
            case 1:
                return PassExporter.FORMAT_PKPASS;

            default:
                return PassExporter.FORMAT_OPENPASS;
        }
    }


}

File: android/src/main/java/org/ligi/passandroid/ui/PassMenuOptions.kt
package org.ligi.passandroid.ui

import android.app.Activity
import android.content.Intent
import android.view.LayoutInflater
import android.view.MenuItem
import androidx.appcompat.app.AlertDialog
import androidx.core.app.NavUtils
import kotlinx.android.synthetic.main.delete_dialog_layout.view.*
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject
import org.ligi.kaxt.startActivityFromClass
import org.ligi.passandroid.R
import org.ligi.passandroid.Tracker
import org.ligi.passandroid.maps.PassbookMapsFacade
import org.ligi.passandroid.model.PassStore
import org.ligi.passandroid.model.Settings
import org.ligi.passandroid.model.pass.Pass
import org.ligi.passandroid.printing.doPrint
import java.io.File

class PassMenuOptions(val activity: Activity, val pass: Pass) : KoinComponent {

    val passStore: PassStore by inject()
    val tracker: Tracker by inject()
    val settings: Settings by inject()

    fun process(item: MenuItem): Boolean {
        when (item.itemId) {

            R.id.menu_delete -> {
                tracker.trackEvent("ui_action", "delete", "delete", null)

                val builder = AlertDialog.Builder(activity)
                builder.setMessage(activity.getString(R.string.dialog_delete_confirm_text))
                builder.setTitle(activity.getString(R.string.dialog_delete_title))
                builder.setIcon(R.drawable.ic_alert_warning)

                val sourceDeleteCheckBoxView = LayoutInflater.from(activity).inflate(R.layout.delete_dialog_layout, null)

                val source = pass.getSource(passStore)
                if (source != null && source.startsWith("file://")) {

                    sourceDeleteCheckBoxView.sourceDeleteCheckbox.text = activity.getString(R.string.dialog_delete_confirm_delete_source_checkbox)
                    builder.setView(sourceDeleteCheckBoxView)
                }

                builder.setPositiveButton(activity.getString(R.string.delete)) { _, _ ->
                    if (sourceDeleteCheckBoxView.sourceDeleteCheckbox.isChecked) {

                        File(source!!.replace("file://", "")).delete()
                    }
                    passStore.deletePassWithId(pass.id)
                    if (activity is PassViewActivityBase) {
                        val passListIntent = Intent(activity, PassListActivity::class.java)
                        NavUtils.navigateUpTo(activity, passListIntent)
                    }
                }
                builder.setNegativeButton(android.R.string.no, null)

                builder.show()

                return true
            }

            R.id.menu_map -> {
                PassbookMapsFacade.startFullscreenMap(activity)
                return true
            }

            R.id.menu_share -> {
                tracker.trackEvent("ui_action", "share", "shared", null)
                PassExportTaskAndShare(activity, passStore.getPathForID(pass.id)).execute()
                return true
            }

            R.id.menu_edit -> {
                tracker.trackEvent("ui_action", "share", "shared", null)
                passStore.currentPass = pass
                activity.startActivityFromClass(PassEditActivity::class.java)
                return true
            }

            R.id.menu_print -> {
                doPrint(activity, pass)
                return true
            }
        }
        return false
    }

}
